gpii/node_modules/flowManager/src/RepositorySolutionsLoader.js

```diff
@@ -0,0 +1,110 @@
+/*!
+GPII Full SHA Revision DataSource
+
+Copyright 2020 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+require("kettle");
+
+fluid.registerNamespace("gpii.flowManager.repositorySolutionsLoader");
+
+fluid.defaults("gpii.flowManager.repositorySolutionsLoader", {
+    gradeNames: ["kettle.dataSource.URL"],
+    urlTemplate: "%prefix/%revision/%suffix/%fileName",
+    url: {
+        expander: {
+            funcName: "fluid.stringTemplate",
+            args: ["{that}.options.urlTemplate", {
+                prefix: "{that}.options.urlPrefix",
+                suffix: "{that}.options.urlSuffix"
+            }]
+        }
+    },
+    termMap: {
+        revision: "%directRevision",
+        fileName: "%directFileName"
+    },
+    // The URL prefix and suffix are distributed from a config file such as
+    // e.g., gpii.flowManager.config.local.base, as is the path to the local
+    // folder in which to save the contents of the downloaded solutions registry
+    // file.
+    urlPrefix: "",
+    urlSuffix: "",
+    solutionsRegistryFolderPath: "",
+    protocol: "https:",  // default
+    members: {
+        // Contents of the solutions registry file downloaded from the source
+        // code repository
+        solutionsRegistry: null
+    },
+    components: {
+        encoding: {
+            type: "kettle.dataSource.encoding.JSON5"
+        }
+    },
+    invokers: {
+        getSolutions: {
+            funcName: "gpii.flowManager.repositorySolutionsLoader.getSolutions",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
```

Comment:
`gpii.flowManager.repositorySolutionsLoader.getSolutions()` function at line 80 accepts 4 parameters. This argument list only passes the first 3. Shall the fourth be added here? Does this also mean the test doesn't cover the fourth argument?

Reply:
Moreover, due to the connection between the platform id and the file name: file-name === platform-id + ".json5". Passing platform id might be good enough.

Reply:
As it happens, the `platformId` is actually part of the `fileName`.  For example, if the solutions registry file to download is `win32.json`, then the `platformId` is `win32`. Passing the platform id is simply a convenience.  I'll re-write `getSolutions()` to get the platform id from the `fileName`.

Reply:
Somehow, I completely missed your excellent suggestion of just passing a `platformId`, and constructing the `fileName` inside of `getSolutions()`.  That's much more intuitive, and I've rewritten it that way.